### PR TITLE
Add nixpkgs-fmt to lint Nix files.

### PR DESCRIPTION
**Pull request by samcraigdev**
- Add linter `nixpkgs-fmt-nixos`
- Add linter `nixpkgs-fmt-alpine`
- Update CHANGELOG.md
- Add working & failing `.nix` examples

Went ahead and added two runners via `nixpkgs-fmt-nixos` and
`nixpkgs-fmt-alpine`:

- Runner `nixpkgs-fmt-nixos` utilizes the `cli_docker_image` linter flag, running image `samcraig/nixpkgs-fmt` to download and install `nixpkgs-fmt` via `nixos/nix`.
- Runner `nixpkgs-fmt-alpine` utilizes the native mega-linter docker installation method. This script is a bit cumbersome because due to how the ergonomics for the Nix profile setup works on the edge installation of `nix` via `apk`.

## Proposed Changes

1. Add [nixpkgs-fmt](https://github.com/nix-community/nixpkgs-fmt) to lint [Nix](https://nixos.org/) files.

### Author/Contributor
- [x] If documentation is needed for this change, has that been included in this pull request

### Reviewing Maintainer
- [ ] Label as `breaking` if this is a large fundamental change
- [x] Label as either `automation`, `bug`, `documentation`, `enhancement`, `infrastructure`, or `performance`

### CODE REVIEW

**Comment by samcraigdev** [comment]
Added two runners here, imagine we'll likely choose one here for ergonomics sake. Would appreciate input on the decision. After we decide on a primary I'll go ahead and modify the descriptor and send in another commit for final review.

**Comment by nvuillam** [comment]
Usually I prefer to have executables directly in the main mega-linter image, but the Dockerfile install instructions for that seem long and heavy, so maybe the use of another docker image would make sense...

Let's first check what CI says and both descriptions :)

**Comment by samcraigdev** [comment]
Looks like the documentation references generated in the root level `README.md` via `bash build.sh --doc` use absolute URLs and not relative URLs, essentially referencing files on the master's origin tree that don't yet exist when I got ahead and run the aforementioned build command:

https://github.com/nvuillam/mega-linter/blob/540d3ded099951ec5b7381b553670b6b7e24b355/.automation/build.py#L38-L40

Not 100% sure on the ergonomics of how / when you create documentation in your contribution workflow, simply following step `#5` via:

https://github.com/nvuillam/mega-linter/blob/540d3ded099951ec5b7381b553670b6b7e24b355/.github/CONTRIBUTING.md#L60-L68

From a quick search it doesn't look like CI is handling this, so I'm assuming it's via a manual build / submission. So, I'll wait on an answer here before I go ahead and send over an updated commit.

**Comment by nvuillam** [comment]
Own Mega-Linter is not an issue, it's normal it crashes for new linter, and if broken links is the only error it's ok ^^

But Docker build failure needs to work :/

image

**Comment by samcraigdev** [comment]
> Own Mega-Linter is not an issue, it's normal it crashes for new linter, and if broken links is the only error it's ok ^^
> 
> But Docker build failure needs to work :/
> 
> image

Ok, makes sense.

That latest commit should address the Docker build error via removal of `nixpkgs-fmt-alpine` in lieu of `nixpkgs-fmt-nixos` (renamed to the default `nixpkgs-fmt`).

**Comment by samcraigdev** [comment]
@nvuillam Bumping, let me know if there are any blockers here for the review.

**Comment by nvuillam** [comment]
It seems the main CI job does not run... that's strange... please can you merge conflicts then run again bash build.sh --doc to see if it triggers again Build job for DEV ?

**Comment by samcraigdev** [comment]
> It seems the main CI job does not run... that's strange... please can you merge conflicts then run again bash build.sh --doc to see if it triggers again Build job for DEV ?

@nvuillam👍

File: megalinter/descriptors/nix.megalinter-descriptor.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+descriptor_id: NIX
+descriptor_type: format
+descriptor_flavors:
+  - all_flavors # Applicable to CI in any language project
+file_extensions:
+  - ".nix"
+linters:
+  - linter_name: nixpkgs-fmt
+    linter_repo: https://github.com/serokell/nixpkgs-fmt
```
Comment (nvuillam) [inline megalinter/descriptors/nix.megalinter-descriptor.yml:9]: isn't it https://github.com/serokell/nixfmt ?

File: megalinter/descriptors/nix.megalinter-descriptor.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    linter_url: https://nixpkgs-fmt.serokell.io/
```
Comment (nvuillam) [inline megalinter/descriptors/nix.megalinter-descriptor.yml:10]: dead link

**Review by nvuillam — CHANGES_REQUESTED** [review]
Just made a quick check and there are dead links ^^

**Pull request closed without merging**